android/src/com/mapswithme/maps/gallery/ImageAdapter.java

```diff
@@ -0,0 +1,81 @@
+package com.mapswithme.maps.gallery;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+import com.mapswithme.maps.R;
+import com.mapswithme.maps.widget.recycler.RecyclerClickListener;
+
+import java.util.ArrayList;
+
+class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.ViewHolder>
+{
+  private final ArrayList<Image> mItems;
+  private final RecyclerClickListener mListener;
+
+  ImageAdapter(ArrayList<Image> images, RecyclerClickListener listener)
+  {
+    mItems = images;
+    mListener = listener;
+  }
+
+  @Override
+  public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
+  {
+    return new ViewHolder(LayoutInflater.from(parent.getContext())
+                                        .inflate(R.layout.item_image, parent, false), mListener);
+  }
+
+  @Override
+  public void onBindViewHolder(ViewHolder holder, int position)
+  {
+    holder.bind(mItems.get(position), position);
+  }
+
+  @Override
+  public int getItemCount()
+  {
+    if (mItems == null)
```

Comment:
I think would be better if mItems was non-null (default initialization with empty list, for example) and null checking would be removed at all. But, if to do this is impossible then I agree with @ygorshenin that ternary operation more cleaner and readable.